[yyyaohhh] iP

yyyaohhh opened the pull request:
### Bocchi · The · Note！

> “It might not resonate with all, but those that it does, it’ll hit deep.” -Yamada Ryō

Bocchi chan helps you with remembering the things that you need to do~(ฅ´ω`ฅ)

- Bocchi remembers well~
- Bocchi is easy to interact with~
- Bocchi is **CUTE**~

To interact with her now, simply:
1. Download it from [here](https://github.com/yyyaohhh/ip).
2. Add in your tasks.
3. Let her manage your tasks for you♥
4. _Remember not to bully her!_ 😉

Features:

- [X] Managing tasks
- [X] Managing deadlines
- [ ] Reminders (coming soon)~

If you Java programmer, you can ~tweak her~ use her to practice Java too. Here's the `main` method:
```
    public static void main(String[] args) {
        new Duke("./data/duke.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

yyyaohhh committed:
level-0

yyyaohhh committed:
Level-1

yyyaohhh committed:
Level-2

yyyaohhh committed:
Level-3

yyyaohhh committed:
Level-4

yyyaohhh committed:
Automated Text UI Testing

yyyaohhh committed:
Level-5

yyyaohhh committed:
Add Level-7

yyyaohhh committed:
Add Dates and Times

yyyaohhh committed:
Added A-MoreOOP

yyyaohhh committed:
Merge remote-tracking branch 'origin/add-gradle-support'

yyyaohhh committed:
Add Gradle

yyyaohhh committed:
Add A-JUnit

yyyaohhh committed:
Add A-JavaDoc

yyyaohhh committed:
Add Level-9

yyyaohhh committed:
Merge branch 'Level-9'

brein62 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,39 @@
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import tasks.Deadline;
```
Seems like the `tasks` package and the Java files is not present in your repository when you refactored the files to improve OOP and add files in packages. You can check if these files are there on your local side and commit them into your repository

brein62 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,39 @@
+import java.io.FileNotFoundException;
```
Should be part of a package like `duke`! If you are using IntelliJ, you can move all the files from `main/java` into `main/java/duke` and IntelliJ will auto refactor it for you - so the file should look like this:

```java
package duke;

import java.io.FileNotFoundException;
// Other code
```

brein62 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,39 @@
+    private Ui ui;
+    private Storage storage;
+    private TaskList tasks;
+    public Duke(String filePath) {
```
Should have a blank line to separate private fields from public methods, as well as public methods from other public methods

brein62 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,39 @@
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
+    }
+    public void run() {
```
Should have a blank line to separate public methods from other public methods. Same applies to the `main()` method below.

In this method, better to leave a few blank lines within the method to make it more readable 😄 , such as
```java
try {
    this.tasks = new TaskList(this.storage.load());
} catch (FileNotFoundException e) {
    System.out.println("File not found");
}

Parser parser = new Parser();
parser.parse(this.tasks);

try {
    // ...
```

yyyaohhh replied:
Thank you very much for the feedback! Will make the relevant amendments! 😀

brein62 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,39 @@
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found");
+        }
+        Parser execute = new Parser();
```
This `Parser` object should be a noun: `parser` or `executor`

brein62 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,113 @@
+                        System.out.println("O...Omedeto! I have... have marked this as done!!:DD");
+                        System.out.println(tasks.printTask(taskNum - 1));
+                        Ui.line();
+
```
Unintended blank line 🤔

brein62 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,113 @@
+                    String[] parts = echo.split(" ");
+                    if (parts[0].equals("mark")) {
+                        if (parts.length == 1) {
+                            throw new DukeException("AAA...AGHHH!!! The number to be marked done... c...cannot be empty!!!°(°ˊДˋ°) °");
```
°(°ˊДˋ°) °

The overall function LGTM! But be careful and make sure the lines don't exceed 120 characters!

brein62 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,94 @@
+import tasks.Event;
+import tasks.ToDo;
+import tasks.Task;
+/**
```
Can try to leave a blank line between the import statements and the JavaDoc for the `TaskList` class!

brein62 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,94 @@
+     * Prints the entire list of tasks with numbering for reference.
+     */
+    public void printList() {
+        for(int i = 0; i < tasks.size(); i++) {
```
Need space between `for` and the area in parentheses:

```suggestion
        for (int i = 0; i < tasks.size(); i++) {
```

brein62 commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,31 @@
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
+    public Ui() {
```
Need to leave a blank space above method definitions!

brein62 reviewed (commented):
Good job! Overall logic of the code LGTM and I really love the error messages you show to the user. Just a few overall suggestions and tips to follow coding standards:

1. For some reason, your `main/java/tasks` folder is missing in this repository. I think they haven't been committed yet so you can try to find them and commit them as needed. It's a bit weird to see test cases for `test/java/tasks` when the actual code isn't even present :thinking:
2. Don't forget to leave a blank line above method definitions like `public static void main(String[] args)` for all methods to improve readability of the code.
3. Don't forget to move the files in the `main/java` folder into `main/java/duke` and include these files in a package. According to coding standards, **all** files should be put in a package, even the main `Duke.java`.
4. If you have time, can try to add JavaDoc onto `Duke.java` and files without JavaDoc but I don't think it's a requirement in `A-JavaDoc` to include JavaDoc onto everything?

yyyaohhh committed:
Add A-Packages

licongshen12 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,97 @@
+    /**
+     * Prints the number of tasks in the list.
+     */
+    public void numOfTask() {
+        System.out.println("N... Now you have... " + this.tasks.size() + " tasks in the list. 0w0");
+    }
```
Maybe you could consider renaming this method so that it starts with a verb?

licongshen12 commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,75 @@
+ */
+public class Task {
+    protected String taskName;
+    protected boolean isDone;
```
I like how your naming of booleans adheres to Java coding standards!

licongshen12 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,133 @@
+        String echo = echoObject.nextLine();
+        while (!echo.equals("bye")) {
+            try {
+                if (!echo.equals("list")) {
```
Perhaps you could consider using switch statements to handle the different cases?

licongshen12 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,133 @@
+            try {
+                if (!echo.equals("list")) {
+                    String[] parts = echo.split(" ");
+                    if (parts[0].equals("mark")) {
```
Maybe you could use an enum to handle the different command types?

licongshen12 reviewed (commented):
Overall, the code functionality is good and the documentation is well done. LGTM!

yyyaohhh committed:
JavaFX Errors

yyyaohhh committed:
JavaFX part 4 error

yyyaohhh committed:
NullPointerException Error

yyyaohhh committed:
Add GUI

yyyaohhh committed:
Add GUI for Duke Chat bot. It is more visually appealing to use a GUI instead of CLI and allows better interactions with users. JavaFX is used for making the GUI. Images of Bocchi and Kita are added to represent the user and the chatbot because this chatbot is designed as Bocchi-chan.

yyyaohhh committed:
Try to merge and remove the duplicated branch

yyyaohhh committed:
Merge pull request #2 from yyyaohhh/GUI

Try to merge and remove the duplicated branch

yyyaohhh committed:
Try to merge and remove the duplicated branch

yyyaohhh committed:
Merge pull request #3 from yyyaohhh/Level-10

Try to merge and remove the duplicated branch

yyyaohhh committed:
Merge pull request #4 from yyyaohhh/branch-Level-10

Branch level 10

yyyaohhh committed:
Add assert feature

Assert feature is used to document important assumptions that should hold at various points in the code and check for the accuracy of the code.

It is added it to the Parser class as the class parse and process the input and gives the output, so its accuracy is very important.

Let's,
* the asserts checks if the methods returns a null.

yyyaohhh committed:
Improve Code Quality

Critically examine the code and refactor to improve the code quality in the Parser class.
The method parse(TaskList tasks, String echo) in the Parser class was too long and had arrow head shape, due to the use of embedded if else statements.
The long method is being broken down into many shorter methods that each does a function, according to SLAP. for example markCase(TaskList tasks, String[] parts) marks the case. The embedded if-else statements are also cleaned to reduce the arrow head effect.

yyyaohhh committed:
Add C-Help

help command: Allow first time users to learn about all the available commands in the chat bot.
It explain in detail about all the commands: what they do and the correct input format.
It is added as an extension feature.

yyyaohhh committed:
Add User Guide

yyyaohhh committed:
Solve read file issue

yyyaohhh committed:
Fix Ui.png

yyyaohhh committed:
Rename Ui.png

yyyaohhh committed:
Fix Ui.png

yyyaohhh committed:
Add Ui.png
